### PR TITLE
Add cryptographic random.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/compiler/propagation/type_primitive_crypto_random.cc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// directory of this repository.
+
+#include "type_primitive.h"
+
+namespace toit {
+namespace compiler {
+
+MODULE_TYPES(crypto_random, MODULE_CRYPTO_RANDOM)
+
+TYPE_PRIMITIVE_ANY(random)
```
Comment (kasperl) [inline src/compiler/propagation/type_primitive_crypto_random.cc:23]:
```suggestion
TYPE_PRIMITIVE_BYTE_ARRAY(random)
```

File: src/primitive_crypto_random.cc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  ByteArray::Bytes bytes(result);
+
+#ifdef TOIT_ESP32
+  // We should eventually try to use std::random_device here too.
+  // https://github.com/espressif/esp-idf/issues/11398
+  esp_fill_random(bytes.address(), size);
+#else
+  // The std::random_device is mapped to /dev/urandom on Linux/macOS and to
+  // a cryptographic API on Windows.
+  std::random_device rd;
```
Comment (kasperl) [inline src/primitive_crypto_random.cc:52]: Maybe avoid the abbreviations here? device and distribution seem better than rd and dis.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/compiler/propagation/type_primitive_crypto_random.cc (hunk 1 above)
Comment (floitsch) [reply src/compiler/propagation/type_primitive_crypto_random.cc]: done.

File: src/primitive_crypto_random.cc (hunk 2 above)
Comment (floitsch) [reply src/primitive_crypto_random.cc]: done.

**Pull request merged**